batch7crud-roll1/src/main/java/com/lftechnology/batch7crud/controller/EmployeeController.java

```diff
@@ -0,0 +1,158 @@
+	}
+
+    private void list(HttpServletRequest request, HttpServletResponse response, int page) throws ServletException, IOException {
+
+        try {
+            List<Employee> employees = employeeService.fetch();
+            request.setAttribute("employees", employees);
+            request.getServletContext().getRequestDispatcher("/WEB-INF/views/employee/list.jsp").forward(request, response);
+        } catch (DataException e) {
+            e.printStackTrace();
+            response.setStatus(HttpServletResponse.SC_INTERNAL_SERVER_ERROR);
+            request.setAttribute("message", e.getMessage());
+            request.getServletContext().getRequestDispatcher("/WEB-INF/views/error.jsp").forward(request, response);
+        }
+    }
+
+    private void create(HttpServletRequest request, HttpServletResponse response) throws ServletException, IOException {
+        request.getServletContext().getRequestDispatcher("/WEB-INF/views/employee/create.jsp").forward(request, response);
+    }
+
+    private void createProcess(HttpServletRequest request, HttpServletResponse response) throws ServletException, IOException {
+        int id = TypeCaster.toInt(request.getParameter("id"));
+        String userName = request.getParameter("userName");
+        String password = request.getParameter("password");
+        String fullName = request.getParameter("fullName");
+        String department = request.getParameter("department");
+        String address = request.getParameter("address");
+        Employee employee = new Employee(id,userName,password,fullName,department,address);
+
+        try {
+            employeeService.create(employee);
+            response.sendRedirect("/employees");
+        } catch (DataException e) {
+            e.printStackTrace();
+            response.setStatus(HttpServletResponse.SC_INTERNAL_SERVER_ERROR);
+            request.setAttribute("message",e.getMessage());
+            request.getServletContext().getRequestDispatcher("/WEB-INF/views/error.jsp").forward(request, response);
+        }
+    }
+
+    private void edit(HttpServletRequest request, HttpServletResponse response, int id) throws ServletException, IOException {
+        try {
+            Employee employee = employeeService.fetchById(id);
+            request.setAttribute("employee", employee);
+            request.getServletContext().getRequestDispatcher("/WEB-INF/views/employee/edit.jsp").forward(request, response);
+        } catch (DataException e) {
+            e.printStackTrace();
+            response.setStatus(HttpServletResponse.SC_INTERNAL_SERVER_ERROR);
+            request.setAttribute("message", e.getMessage());
+            request.getServletContext().getRequestDispatcher("/WEB-INF/views/error.jsp").forward(request, response);
+        }
+    }
+
+    private void editProcess(HttpServletRequest request, HttpServletResponse response,int id) throws ServletException, IOException {
+        id = TypeCaster.toInt(request.getParameter("id"));
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Introduce a new variable instead of reusing the parameter "id". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1226)